src/Aspire.Hosting.Azure.Functions/AzureFunctionsProjectResourceExtensions.cs

```diff
@@ -93,6 +95,29 @@ public static class AzureFunctionsProjectResourceExtensions
             });
     }
 
+    /// <summary>
+    /// Configures the Azure Functions project resource to use the specified port as its HTTP endpoint.
+    /// This method queries the launch profile of the project to determine the port to
+    /// use based on the command line arguments configure in the launch profile,
+    /// </summary>
+    /// <param name="builder">The resource builder for the Azure Functions project resource.</param>
+    /// <returns>An <see cref="IResourceBuilder{AzureFunctionsProjectResource}"/> for the Azure Functions project resource with the endpoint configured.</returns>
+    private static IResourceBuilder<AzureFunctionsProjectResource> WithFunctionsHttpEndpoint(this IResourceBuilder<AzureFunctionsProjectResource> builder)
+    {
+        var launchProfile = builder.Resource.GetEffectiveLaunchProfile();
```

Comment:
We don't need to block this PR but I think we'll also want to figure out how to let support the entire launch profile (application url etc).